src/metabase/automagic_dashboards/populate.clj

```diff
@@ -25,28 +29,92 @@
       :color       color
       :description description)))
 
-(def automagic-collection
-  "Get or create collection used to store all autogenerated cards.
-   Value is wrapped in a delay so that we don't hit the DB out of order."
-  (delay
-   (or (db/select-one 'Collection
-         :name "Automatically Generated Questions")
-       (create-collection! "Automatically Generated Questions"
-                           "#509EE3"
-                           "Cards used in automatically generated dashboards."))))
+(defn automagic-collection
+  "Get or create collection used to store all autogenerated cards."
+  []
+  (or (db/select-one 'Collection
+        :name     [:like "Automatically Generated Questions%"]
+        :archived false)
+      (let [c (db/count 'Collection
+                :name [:like "Automatically Generated Questions%"])]
+        (create-collection!
+         (if (pos? c)
+           (format "Automatically Generated Questions %s" (inc c))
+           "Automatically Generated Questions")
+         "#509EE3"
```

Comment:
I'm not sure what this is. Probably would be easiest to def it as a constant and put a comment on it.

Reply:
It's done like this so that the user can delete the autogenerated collection and it pops back up next time they save an xray.